src/countAnimals.js

```diff
@@ -1,7 +1,22 @@
-const data = require('../data/zoo_data');
+const { species } = require('../data/zoo_data');
+
+function defaultAnimals() {
+  const output = {};
+
+  species.forEach(({ name, residents }) => {
+    output[name] = residents.length;
+  });
+
+  return output;
+}
 
 function countAnimals(animal) {
-  // seu código aqui
+  if (!animal) return defaultAnimals();
+
+  const { residents } = species.find((specie) => specie.name === animal.specie);
+  const filteredByGender = residents.filter((resident) => resident.sex === animal.sex).length;
```

Comment:
```suggestion
  const filteredByGender = residents.filter((resident) => resident.sex === animal.sex).length;
```

Talvez esse nome deva indicar que essa variável guarda somente a quantidade

Reply:
Faz sentido! Vou aplicar!